control-announcements/src/main/java/org/oskari/announcements/helpers/AnnouncementsDBHelper.java

```diff
@@ -0,0 +1,367 @@
+                row.put("begin_date", df.format(rs.getDate("begin_date")));
+                row.put("end_date", df.format(rs.getDate("end_date")));
+                row.put("active", rs.getBoolean("active"));
+                results.put(row);
+            }
+
+        } catch (SQLException e) {
+            LOG.error(e, "Cannot create SQL query, sql=" + sqlWithParams);
+
+        } finally {
+            if(conn != null) {
+                try {
+                    conn.close();
+                } catch (Exception e) {
+                    LOG.error(e);
+                }
+            }
+        }
+
+        JSONObject json = new JSONObject();
+        json.put("data", results);
+
+        return json;
+
+    }
+
+    /**
+     * Handle saving announcement to database
+     * @param params
+     * @return
+     * @throws JSONException
+     * @throws ActionParamsException
+     */
+    public static JSONObject saveAnnouncement(ActionParameters params) throws JSONException, ActionParamsException {
+        Connection conn = null;
+
+        JSONArray results = new JSONArray();
+        String sql = "";
+        String sqlWithParams = "";
+
+        try  {
+        
+            conn = getConnection();
+            StringBuilder sb = new StringBuilder();
+            sb.append("INSERT INTO oskari_announcements(title, content, begin_date, end_date, active) ");
+            sb.append("VALUES (? ::VARCHAR, ? ::VARCHAR, ? ::DATE, ? ::DATE, ? ::BOOLEAN) ");
+            sb.append("RETURNING id;");
+
+            
+
+            sql = sb.toString();
+            List<AnnouncementParams> announcementParams = AnnouncementsParser.parseAnnouncement(params);
+
+            PreparedStatement pstmt = conn.prepareStatement(sql);
+
+            int index = 1;
+            for (int i=0; i<announcementParams.size(); i++) {
+                AnnouncementParams announcementParam = announcementParams.get(i);
+                if (announcementParam.getValue() instanceof Integer) {
+                    pstmt.setInt(index, (int)announcementParam.getValue());
+                    index++;
+                } else if (announcementParam.getValue() instanceof String) {
+                    pstmt.setString(index, (String)announcementParam.getValue());
+                    index++;
+                } else if (announcementParam.getValue() instanceof Date) {
+                    pstmt.setDate(index, (Date)announcementParam.getValue());
+                    index++;
+                } else if (announcementParam.getValue() instanceof Boolean) {
+                    pstmt.setBoolean(index, (Boolean)announcementParam.getValue());
+                    index++;
+                }
+            }
+
+            ResultSet rs = pstmt.executeQuery();
+            while (rs.next()) {
+                results.put(rs.getInt("id"));
+            }
+
+        } catch (SQLException e) {
+            LOG.error(e, "Cannot create SQL query, sql=" + sqlWithParams);
```

Comment:
Isn't `sqlWithParams` always empty?

Reply:
It would be a much more standard way of detecting errors if you throw an exception if there is one. Now the caller needs to check if it received an id or not from a JSON returned from this method.